spec/support/network_helpers.rb

```diff
@@ -52,20 +52,22 @@ def check_availability_by_http_request(host, port)
     false
   end
 
-  # Adds the DD specific tracer configuration environment to the inputted tracer headers as
-  # a comma separated string of `k=v` pairs.
+  # Gets the Datadog Trace Configuration and other relevant data and returns a comma separated string of key/value pairs.
   #
-  # @return [Hash] trace headers
-  def parse_tracer_config_and_add_to_headers(trace_headers)
-    dd_service = Datadog.configuration.service
-    dd_span_attribute_schema = 'v0'
-    trace_variables = ENV.to_h.select { |key, _| key.start_with?('DD_') }.map { |key, value| "#{key}=#{value}" }.join(',')
-    if trace_variables.empty?
-      trace_variables = "DD_SERVICE=#{dd_service},DD_SPAN_ATTRIBUTE_SCHEMA=#{dd_span_attribute_schema}"
-    else
-      trace_variables += ",DD_SERVICE=#{dd_service},DD_SPAN_ATTRIBUTE_SCHEMA=#{dd_span_attribute_schema}"
+  # @return [String] Key/Value pairs representing relevant Tracer Configuration
+  def parse_tracer_config(trace)
+    dd_env_variables = ENV.to_h.select { |key, _| key.start_with?('DD_') }
+    trace.spans.each do |s|
+      if s && s.meta && s.meta['_expected_service_name'] && s.meta['_remove_integration_service_names_enabled'] == 'true'
+        dd_env_variables['DD_SERVICE'] = s.meta['_expected_service_name']
+        dd_env_variables['DD_TRACE_SPAN_ATTRIBUTE_SCHEMA'] = 'v1'
```

Comment:
Because this is running in a `trace.spans` loop, `dd_env_variables['DD_SERVICE']` will be overwritten multiple times, with the last span winning. Is that expected?

Reply:
Updated this to not overwrite service name if a valid service name has already been found.